### PR TITLE
Revamp Log4j tests

### PR DESCRIPTION
**Pull request by vy**
I am an Apache Logging Services (the Apache Software Foundation _project_ responsible for Log4j, Log4cxx, Log4net, etc. _subprojects_) PMC member – see [the team](https://logging.apache.org/team-list.html). This PR revamps the Log4j tests with following changes:

- **Replaced existing tests<sup>1</sup>** with new, improved ones in [the `fuzzing` branch<sup>2</sup> of the official Log4j repository](/apache/logging-log4j2/compare/2.x..fuzzing). This will not only ensure that tests will be maintained as a part of the official project, also enables maintainers to introduce new tests, make changes, etc. without touching the OSS-Fuzz code base.

- **Introduced JSON dictionary and seed corpus** for fuzzing JSON-related Log4j components

- **Rewrote `Dockerfile`** to match the JDK version to the one used by the official Log4j project.

- **Delegated `build.sh`** to [the one provided in the official Log4j distribution](/apache/logging-log4j2/blob/fuzzing/oss-fuzz-build.sh). This enables maintainers to introduce new tests, make changes, etc. without touching the OSS-Fuzz code base.

- **Rewrote `project.yaml`** such that
  - Interested PMC members were added as `primary_contact`
  - Removed existing `vendor_ccs`, since found vulnerabilities should not be disclosed to individuals outside the PMC

- **Renamed `log4j2` project folder to `apache-logging-log4j2`** to comply with the naming convention followed by other ASF projects integrated with OSS-Fuzz

<sup>1</sup> Existing tests were not even compiling anymore due to bit rot.
<sup>2</sup> I choose using a dedicated branch to carry out the development. Once I have a ClusterFuzz account and see things moving, I will merge the `fuzzing` branch to `2.x`, the branch serving Log4j 2 releases.

### CODE REVIEW

**Review by fmeum — APPROVED** [review]
LGTM

I also looked at the fuzz tests on the fuzzing branch in the log4j repo and they look much, much better than the basic tests I added as part of the initial integration. Great work! You may want to apply for an [integration reward](https://google.github.io/oss-fuzz/getting-started/integration-rewards/).

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,10 @@
+.idea
```
[the remark this answers is not in the document]
Comment (vy) [reply .gitignore]: Thanks for the tip! Removed non-project changes in f2b616af95addc722c048d8711c5130c31366325.

**Comment by vy** [comment]
Hey @DavidKorczynski! Thanks for [triggering the CI](/google/oss-fuzz/actions/runs/10217976863). There, out of 11, only [the `build (libfuzzer, address, x86_64)` check has failed](https://github.com/google/oss-fuzz/actions/runs/10217976863/job/28359136173#logs), but the error message doesn't make much sense to me. (Could it be because of I renamed `/projects/log4j2` to `/projects/apache-logging-log4j2`?) Do you mind helping out, please?

**Review by fmeum — APPROVED** [review]
@DavidKorczynski Could you review for approval?

**Review by DavidKorczynski — APPROVED** [review]
Thanks @vy and @fmeum !!

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
> Renamed log4j2 project folder to apache-logging-log4j2 to comply with the naming convention followed by other ASF projects integrated with OSS-Fuzz

Could we avoid doing this please, and keep it in the same folder? Due to the backend running OSS-Fuzz we prefer to not change the name

File: projects/log4j2/build.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash -eu
-# Copyright 2021 Google Inc.
+# Copyright 2024 Google LLC
```
Comment (DavidKorczynski) [inline projects/log4j2/build.sh:2]: No need to update copyright year (can keep the LLC change)

File: projects/log4j2/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2021 Google LLC
+# Copyright 2024 Google LLC
```
[the remark this answers is not in the document]
Comment (vy) [reply projects/log4j2/Dockerfile]: Fixed in 11d428e5054012dd9944908fe69dfd99ab6d4166.

File: projects/log4j2/build.sh (hunk 2 above)
Comment (vy) [reply projects/log4j2/build.sh]: Fixed in 11d428e5054012dd9944908fe69dfd99ab6d4166.

**Review by DavidKorczynski — APPROVED** [review]

**Pull request merged**